llvm/lib/Target/DirectX/DXILRootSignature.cpp

```diff
@@ -48,6 +48,71 @@ static bool reportValueError(LLVMContext *Ctx, Twine ParamName,
   return true;
 }
 
+// Template function to get formatted type string based on C++ type
+template <typename T> std::string getTypeFormatted() {
```

Comment:
Looks like this could be a StringRef.
```suggestion
template <typename T> StringRef getTypeFormatted() {
```